jerry-core/parser/js/js-lexer.c

```diff
@@ -2290,8 +2329,17 @@ lexer_expect_object_literal_id (parser_context_t *context_p, /**< context */
       {
         lexer_skip_spaces (context_p);
 
-        if (context_p->source_p < context_p->source_end_p
-            && context_p->source_p[0] != LIT_CHAR_COLON)
+#ifndef CONFIG_DISABLE_ES2015_SHORTHAND_OBJ_LITERAL_NOTATION
+        bool not_end_of_literal = (context_p->source_p < context_p->source_end_p
+          && context_p->source_p[0] != LIT_CHAR_COLON
+          && context_p->source_p[0] != LIT_CHAR_RIGHT_BRACE
+          && context_p->source_p[0] != LIT_CHAR_COMMA);
+#else /* !CONFIG_DISABLE_ES2015_SHORTHAND_OBJ_LITERAL_NOTATION */
+        bool not_end_of_literal = (context_p->source_p < context_p->source_end_p
+          && context_p->source_p[0] != LIT_CHAR_COLON);
+#endif /* CONFIG_DISABLE_ES2015_SHORTHAND_OBJ_LITERAL_NOTATION */
+
+        if (not_end_of_literal)
```

Comment:
Please add test for the pre-parser such as:
```
switch (0) { default:
  var a = { foo, foo() {}, bar:5 }
}
```

Reply:
This should throw a `ReferenceError` because `foo` isn't defined initially. Is that intended?

Reply:
The main reason of this test to check, whether the scanner is able to pre-scan the new syntax element. As it seems it can so the thrown `Reference Error` is correct.